qiskit/visualization/circuit/matplotlib.py

```diff
@@ -511,11 +511,20 @@ def _get_layer_widths(self, node_data, wire_map, glob_data):
                         # Update the wire_map with the qubits from the inner circuit
                         flow_wire_map = {
                             inner: wire_map[outer]
-                            for outer, inner in zip(self._qubits, circuit.qubits)
+                            for outer, inner in zip(node.qargs, circuit.qubits)
                             if inner not in wire_map
                         }
+                        flow_wire_map.update(
+                            {
+                                inner: wire_map[outer]
+                                for outer, inner in zip(node.cargs, circuit.clbits)
+                                if inner not in wire_map
+                            }
+                        )
                         if not flow_wire_map:
                             flow_wire_map = wire_map
+                        else:
+                            flow_wire_map.update(wire_map)
```

Comment:
```suggestion
                        flow_wire_map.update(wire_map)
```

Reply:
Tbh, I'm not sure why this update is necessary; when entering an inner scope, the only bits that I think should be accessible are the ones defined by the bit-binding relations in lines 511 to 523.

Reply:
In this circuit, the qc2 circuit can use either existing `qr, cr` or new `qr2, cr2`. If for example `cr` is used, the inner map will not contain this since it's already in the `wire_map`. This means without the combination above, it won't be available for the condition code.
```
qr = QuantumRegister(3, "q")
cr = ClassicalRegister(2, "cr")
qc = QuantumCircuit(qr, cr)
qc.h(0)

qr2 = QuantumRegister(3, "qr2")
cr2 = ClassicalRegister(2, "cr2")
qc2 = QuantumCircuit(qr, cr) # Can be (qr2, cr2)
qc2.x(1).c_if(cr[0], 1) # If cr2, change to cr2

with qc.if_test((cr[0], 1)):
    qc.x(0)
    qc.if_else((cr[1], 1), qc2, None, [0, 1, 2], [0, 1])
qc.y(0).c_if(cr[0], 0)
qc.draw("mpl", style={"showindex": True})
```

Reply:
If you make the changes suggested in your comments there, you'd have made an invalid circuit; it's not permitted to use a register condition that's not defined in the outermost scoped circuit.  It's an unfortunate restriction and a gotcha, but it's required at the moment because we don't have any way of defining scoped storage locations for hardware, nor any way of defining a "binding" of registers from an outer circuit to an inner scope in the same way that we do for bits.

Reply:
Yeah, there's no error checking of it; I guess we'd have to put a special case into `QuantumCircuit.append` to do that (it's not enough to put it in `if_test` / `if_else` because `append` is a valid entry point too, but `_append`'s contract is that the user is explicitly responsible for all error checking).  Maybe that would actually be a good idea - it shouldn't be too expensive, and it would help prevent common issues.